components/server/src/github/github-repository-provider.ts

```diff
@@ -172,28 +172,57 @@ export class GithubRepositoryProvider implements RepositoryProvider {
     }
 
     async getUserRepos(user: User): Promise<string[]> {
-        // Hint: Use this to get richer results:
-        //   node {
-        //       nameWithOwner
-        //       shortDescriptionHTML(limit: 120)
-        //       url
-        //   }
         const result: any = await this.githubQueryApi.runQuery(
             user,
             `
-            query {
+            fragment Repos on RepositoryConnection {
+                nodes {
+                  url
+                }
+              }
+
+              query topRepositories {
                 viewer {
-                    repositoriesContributedTo(includeUserRepositories: true, first: 100) {
-                        edges {
-                            node {
-                                url
-                            }
-                        }
-                    }
+                  contributedTo: repositoriesContributedTo(
+                    first: 100
+                    orderBy: {field: PUSHED_AT, direction: DESC}
+                    includeUserRepositories: true
+                    contributionTypes: [COMMIT]
+                  ) {
+                    ...Repos
+                  }
+                  original: repositories(
+                    first: 100
+                    ownerAffiliations: OWNER
+                    privacy: PUBLIC
```

Comment:
Is this by design that we only take public repos? Any benefits in doing that?

Reply:
Nice catch. Going to drop that line.